tests/Util/PhpServicesCreatorTest.php

```diff
@@ -0,0 +1,140 @@
+            ->in(__DIR__.'/fixtures/YamlConvert/')
+            ->exclude(['FakeClass', 'fake_files', 'container_built_differently', 'source/fixtures_load']);
+
+        foreach ($finder as $file) {
+            $nameTestCase = str_replace('.php', '', $file->getBasename());
+
+            yield $nameTestCase => [
+                file_get_contents($file->getPath().'/source/'.$nameTestCase.'.yaml'),
+                file_get_contents($file->getPath().'/'.$file->getFilename()),
+                $nameTestCase,
+            ];
+        }
+    }
+
+    public function testParameters()
+    {
+        $source = file_get_contents(__DIR__.'/fixtures/YamlConvert/source/parameters.yaml');
+        $expectedSource = file_get_contents(__DIR__.'/fixtures/YamlConvert/parameters.php');
+
+        $creator = new PhpServicesCreator();
+
+        $this->assertSame($expectedSource, $creator->convert($source));
+    }
+
+    /*
+     * Not the same configuration, only test if they're built the same way.
+     */
+    public function testImports()
+    {
+        $yamlContainerBuilder = new ContainerBuilder();
+        $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source/fixtures_load'));
+        $yamlLoader->load('services_imports.yaml');
+
+        $phpContainerBuilder = new ContainerBuilder();
+        $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source/fixtures_load'));
+        $phpLoader->load('services_imports.php');
+
+        $this->assertTrue($yamlContainerBuilder->getParameterBag() == $phpContainerBuilder->getParameterBag());
+        $this->assertTrue($yamlContainerBuilder->getDefinitions() == $phpContainerBuilder->getDefinitions());
+        $this->assertTrue($yamlContainerBuilder->getAliases() == $phpContainerBuilder->getAliases());
+        $this->assertTrue($yamlContainerBuilder->getServiceIds() == $phpContainerBuilder->getServiceIds());
+    }
+
+    public function testLoadResources()
+    {
+        $yamlContainerBuilder = new ContainerBuilder();
+        $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source/fixtures_load'));
+        $yamlLoader->load('load_ressources.yaml');
+
+        $phpContainerBuilder = new ContainerBuilder();
+        $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source/fixtures_load'));
+        $phpLoader->load('load_ressources.php');
+
+        $this->assertTrue($yamlContainerBuilder->getParameterBag() == $phpContainerBuilder->getParameterBag());
+        $this->assertTrue($yamlContainerBuilder->getDefinitions() == $phpContainerBuilder->getDefinitions());
+        $this->assertTrue($yamlContainerBuilder->getAliases() == $phpContainerBuilder->getAliases());
+        $this->assertTrue($yamlContainerBuilder->getServiceIds() == $phpContainerBuilder->getServiceIds());
+    }
```

Comment:
Same: can this be done in the one, main test method?

Reply:
Resolved in slack.